js/script.js

```diff
@@ -0,0 +1,135 @@
+const playerFactory = (name, symbol) => ({ name, symbol });
+
+const board = (() => {
+  const boardArray = ['', '', '', '', '', '', '', '', ''];
+
+  const checkWin = (currentPlayer) => {
+    let ans = false;
+    const WINNINGCOMBINATION = [
+      [0, 1, 2],
+      [3, 4, 5],
+      [6, 7, 8],
+      [0, 3, 6],
+      [1, 4, 7],
+      [2, 5, 8],
+      [0, 4, 8],
+      [2, 4, 6],
+    ];
+
+    WINNINGCOMBINATION.forEach((arr) => {
+      if (
+        boardArray[arr[0]] === currentPlayer.symbol
+        && boardArray[arr[1]] === currentPlayer.symbol
+        && boardArray[arr[2]] === currentPlayer.symbol
+      ) {
+        ans = true;
+      }
+    });
+
+    return ans;
+  };
+
+  const checkDraw = (currentPlayer) => {
+    let ans = false;
+    let count = 0;
+
+    boardArray.forEach((x) => {
+      if (x === 'X' || x === 'O') {
+        count += 1;
+      }
+    });
+
+    if (!checkWin(currentPlayer) && count === 9) {
+      ans = true;
+    }
+
+    return ans;
+  };
+
+  return { boardArray, checkWin, checkDraw };
+})();
+
+const playGame = () => {
+  let playerOneName;
+  let playerTwoName;
+  const startButton = document.getElementById('startButton');
+  const boardDiv = document.getElementById('board');
+  const resetButton = document.getElementById('resetButton');
+  let playerOne;
+  let playerTwo;
+  let currentPlayer;
+
+  function switchCurrentPlayer(boardArray) {
+    let count = 0;
+
+    for (let i = 0; i < boardArray.length; i += 1) {
+      if (boardArray[i] === 'X' || boardArray[i] === 'O') {
+        count += 1;
+      }
+    }
+
+    if (count % 2 === 0) {
+      currentPlayer = playerOne;
+    } else {
+      currentPlayer = playerTwo;
+    }
+  }
+
+  function displayPlayerTurn(currentPlayer) {
+    document.getElementById(
+      'playerTurn',
+    ).innerHTML = `${currentPlayer.name}, it is your turn.`;
+  }
+
+  boardDiv.addEventListener('click', (e) => {
+    if (e.target.innerHTML === '') {
+      const index = e.target.getAttribute('data-index');
+      e.target.innerHTML = currentPlayer.symbol;
+      board.boardArray[index] = currentPlayer.symbol;
+
+      if (board.checkWin(currentPlayer)) {
```

Comment:
- [ ] **[OPTIONAL]** Let's remove event listeners after game is draw or win, to prevent the player from changing cell values. [1](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener)